src/app/friends/friends-add-edit/friends-add-edit.component.ts

```diff
@@ -45,9 +55,14 @@ export class FriendsAddEditComponent implements OnInit {
     return this.addFriendForm.get('friendName');
   }
 
-  // Submit method
+  // Submit method for invite friend
   onSubmit(): void{
-    this.addFriendForm.reset();
-    this.router.navigate(['friends/friends-list']);
+    this.loading = true;
+    this.friendService.postFriend(this.addFriendForm.value).subscribe( response => {
```

Comment:
what id API throws an error> handle errors too

Reply:
example:
` public getRepos() {
    this.loading = true;
    this.errorMessage = "";
    this.githubService.getReposCatchError(this.userName)
      .subscribe(
        (response) => {                           //Next callback
          console.log('response received')
          this.repos = response;
        },
        (error) => {                              //Error callback
          console.error('error caught in component')
          this.errorMessage = error;
          this.loading = false;
    
          //throw error;   //You can also throw the error to a global error handler
        }
      )
  }`

Reply:
DO not use console log. instead use bootstrap toaster to show success and error message

Reply:
di toaster is not working so I am using alert box.

Reply:
Okay for now but please check why toaster is not working. DM me the problem